Transfer/Sources/CreateTransferPresenter.swift

```diff
@@ -0,0 +1,236 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#if !COCOAPODS
+import Common
+import TransferMethodRepository
+import TransferRepository
+import UserRepository
+#endif
+import HyperwalletSDK
+
+protocol CreateTransferView: class {
+    typealias SelectItemHandler = (_ value: HyperwalletTransferMethod) -> Void
+    typealias MarkCellHandler = (_ value: HyperwalletTransferMethod) -> Bool
+
+    func hideLoading()
+    func notifyTransferCreated(_ transfer: HyperwalletTransfer)
+    func showCreateTransfer()
+    func showError(_ error: HyperwalletErrorType, _ retry: (() -> Void)?)
+    func showGenericTableView(items: [HyperwalletTransferMethod],
+                              title: String,
+                              selectItemHandler: @escaping SelectItemHandler,
+                              markCellHandler: @escaping MarkCellHandler)
+    func showLoading()
+    func showScheduleTransfer(_ transfer: HyperwalletTransfer)
+    func updateTransferSection()
+}
+
+final class CreateTransferPresenter {
+    private unowned let view: CreateTransferView
+
+    private lazy var userRepository: UserRepository = {
+        UserRepositoryFactory.shared.userRepository()
+    }()
+
+    private lazy var transferRepository: TransferRepository = {
+        TransferRepositoryFactory.shared.transferRepository()
+    }()
+
+    private lazy var transferMethodRepository: TransferMethodRepository = {
+        TransferMethodRepositoryFactory.shared.transferMethodRepository()
+    }()
+
+    private(set) var clientTransferId: String
+    private(set) var sectionData = [CreateTransferSectionData]()
+    private(set) var availableBalance: String?
+
+    private var sourceToken: String?
+
+    var selectedTransferMethod: HyperwalletTransferMethod!
+    var amount: String?
+    var notes: String?
+    var transferAllFundsIsOn: Bool = false {
+        didSet {
+            guard let availableBalance = availableBalance else {
+                return
+            }
+            amount = transferAllFundsIsOn ? availableBalance : nil
+            view.updateTransferSection()
+        }
+    }
+    var destinationCurrency: String? {
+        return selectedTransferMethod.transferMethodCurrency
+    }
+
+    init(_ clientTransferId: String, _ sourceToken: String?, view: CreateTransferView) {
+        self.clientTransferId = clientTransferId
+        self.sourceToken = sourceToken
+        self.view = view
+    }
+
+    func initializeSections() {
+        sectionData.removeAll()
+
+        let createTransferDestinationSection = CreateTransferSectionDestinationData(
+            isTransferMethodAvailable: selectedTransferMethod != nil
+        )
+        sectionData.append(createTransferDestinationSection)
+
+        if let currency = selectedTransferMethod?.transferMethodCurrency {
+            let createTransferUserInputSection = CreateTransferSectionTransferData(
```

Comment:
rename it to createTransferSectionTransferData instead of createTransferUserInputSection

Reply:
fixed